sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-text.service.ts

```diff
@@ -115,6 +133,100 @@ export class ApplicationWithdrawalImportTextService {
     });
   }
 
+  /**
+   * Generates the application bulk withdrawal validation model.
+   * @param textValidations text validation models to be converted to the validation models.
+   * @param originator institution that sent the file.
+   * @param institutionId institution id.
+   * @returns application bulk withdrawal models to be validated and persisted.
+   */
+  async generateValidationModels(
+    textValidations: ApplicationWithdrawalTextValidationResult[],
+    originator: string,
+    institutionId: number,
+  ): Promise<ApplicationBulkWithdrawalValidationModel[]> {
+    const applicationDataMap: ApplicationDataMap = {};
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationValidationData = await this.getApplicationValidationData(
+      applicationNumbers,
+      institutionId,
+    );
+    applicationValidationData.forEach((record) => {
+      applicationDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
+      };
+    });
+    return textValidations.map((textValidation) => {
+      const validationModel = {} as ApplicationBulkWithdrawalValidationModel;
+      const applicationData =
+        applicationDataMap[textValidation.textModel.applicationNumber];
+      validationModel.sin = textValidation.textModel.sin;
+      validationModel.applicationNumber =
+        textValidation.textModel.applicationNumber;
+      validationModel.withdrawalDate = textValidation.textModel.withdrawalDate;
+      validationModel.applicationFound = false;
+      if (applicationData) {
+        validationModel.applicationFound = true;
+        validationModel.studentSINMatch =
+          textValidation.textModel.sin === applicationData.sin;
+        validationModel.hasCorrectInstitutionCode =
```

Comment:
Are we suppose to do further validation, when originator is incorrect?

Reply:
I just updated this part of the code.